src/EnergyPlus/SystemAvailabilityManager.cc

```diff
@@ -1780,38 +1743,38 @@ namespace SystemAvailabilityManager {
-            for (auto &e : state.dataSystemAvailabilityManager->NCycSysAvailMgrData)
+        if (allocated(state.dataSystemAvailabilityManager->NightCycleData))
+            for (auto &e : state.dataSystemAvailabilityManager->NightCycleData)
                 e.AvailStatus = NoAction;
-        if (allocated(state.dataSystemAvailabilityManager->NVentSysAvailMgrData))
-            for (auto &e : state.dataSystemAvailabilityManager->NVentSysAvailMgrData)
+        if (allocated(state.dataSystemAvailabilityManager->NightVentData))
+            for (auto &e : state.dataSystemAvailabilityManager->NightVentData)
                 e.AvailStatus = NoAction;
-        if (allocated(state.dataSystemAvailabilityManager->DiffTSysAvailMgrData))
-            for (auto &e : state.dataSystemAvailabilityManager->DiffTSysAvailMgrData)
+        if (allocated(state.dataSystemAvailabilityManager->DiffThermoData))
+            for (auto &e : state.dataSystemAvailabilityManager->DiffThermoData)
                 e.AvailStatus = NoAction;
-        if (allocated(state.dataSystemAvailabilityManager->HiTurnOffSysAvailMgrData))
-            for (auto &e : state.dataSystemAvailabilityManager->HiTurnOffSysAvailMgrData)
+        if (allocated(state.dataSystemAvailabilityManager->HiTurnOffData))
+            for (auto &e : state.dataSystemAvailabilityManager->HiTurnOffData)
                 e.AvailStatus = NoAction;
-        if (allocated(state.dataSystemAvailabilityManager->HiTurnOnSysAvailMgrData))
-            for (auto &e : state.dataSystemAvailabilityManager->HiTurnOnSysAvailMgrData)
+        if (allocated(state.dataSystemAvailabilityManager->HiTurnOnData))
+            for (auto &e : state.dataSystemAvailabilityManager->HiTurnOnData)
                 e.AvailStatus = NoAction;
-        if (allocated(state.dataSystemAvailabilityManager->LoTurnOffSysAvailMgrData))
-            for (auto &e : state.dataSystemAvailabilityManager->LoTurnOffSysAvailMgrData)
+        if (allocated(state.dataSystemAvailabilityManager->LoTurnOffData))
+            for (auto &e : state.dataSystemAvailabilityManager->LoTurnOffData)
                 e.AvailStatus = NoAction;
-        if (allocated(state.dataSystemAvailabilityManager->LoTurnOnSysAvailMgrData))
-            for (auto &e : state.dataSystemAvailabilityManager->LoTurnOnSysAvailMgrData)
+        if (allocated(state.dataSystemAvailabilityManager->LoTurnOnData))
+            for (auto &e : state.dataSystemAvailabilityManager->LoTurnOnData)
                 e.AvailStatus = NoAction;
-        if (allocated(state.dataSystemAvailabilityManager->OptStartSysAvailMgrData)) {
-            for (auto &e : state.dataSystemAvailabilityManager->OptStartSysAvailMgrData) {
+        if (allocated(state.dataSystemAvailabilityManager->OptimumStartData)) {
+            for (auto &e : state.dataSystemAvailabilityManager->OptimumStartData) {
                 e.AvailStatus = NoAction;
                 e.isSimulated = false;
             }
```

Comment:
Wish I could change this to a loop.

Reply:
You will be able to very shortly. The struct hierarchy refactoring will enable that.